### ISSUE TITLE
Access token wil not refresh for Oauth2 password clients - SuiteCRM v7.13.3+

### ISSUE THREAD

**Issue by scottfrancismiller** [salesagility/SuiteCRM#10268]
#### Issue

Applications will not receive a refreshed access token for Oauth2 password client

#### Expected Behavior

Access token should be refreshed upon request from application using refresh token.

#### Actual Behavior

Access token is not refreshed and applications are required to re-authenticate after access token expiration.

#### Possible Fix

This possibly stems from changes in merged commit in PR https://github.com/salesagility/SuiteCRM/pull/9965

#### Steps to Reproduce

1. Install fresh copy of SuiteCRM 7.13.3+
2. Generate openssl keys 
3. Create new Oauth2 password client
4. **Get New Access Token** using Postman
5. Attempt to **Refresh** access token
6. Postman console error upon  **Refresh**  attempt: ```Error: invalid_client, Description: Client authentication failed```

#### Context

Since upgrading to SuiteCRM 7.13.3+ our [SuiteCRM Outlook Plugin](https://github.com/salesagility/SuiteCRM-Outlook-Plugin) users need to re-authenticate every hour after their access tokens expire. 
I can verify that access tokens can be successfully refreshed in SuiteCRM 7.13.0 - 7.13.2. 
This appears to be an issue in SuiteCRM 7.13.3 and later.

#### Your Environment

* SuiteCRM Version used: 7.14.1
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox 120
* Environment name and version (e.g. MySQL, PHP 7): MySQL 8.0, PHP 8.2
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 22.04

**Comment by ShadMickelberry** [comment]
I can confirm this is still an issue on 7.14.3. After much searching and hair pulling I'm glad to find this issue and it isn't just me. Have you been able to implement a workaround by manually reverting the files in the linked pull request?

**Comment by scottfrancismiller** [comment]
Thanks for confirming. I could not get past this by manually reverting the merged files.

In my experience there is a failure to get past the `validateClient` function in `Api/V8/OAuth2/Repository/ClientRepository.php` where the `$client->allowed_grant_type` checks against the `$grantType` when the value is "refresh_token" - The only possible `allowed_grant_type` values for an `oauth2clients` record appear to be "password" or "client_credentials"

I'm sure there is more to it than that because simply adding an `|| @grantType === 'refresh_token'` clause the `$client->allowed_grant_type` check only satisfies a Refresh Token request for Postman, the SuiteCRM Outlook Plugin still fails to refresh the access token.

Ultimately, we've had to increase the access token expiration timer as a temporary workaround until this is fixed.

**Comment by ShadMickelberry** [comment]
Thank you @scottfrancismiller. I was coming to the same conclusion in Api/V8/OAuth2/Repository/ClientRepository.php. I just tested out removing the check on $grantType but that did not resolve the issue.

**Comment by ShadMickelberry** [comment]
@scottfrancismiller @johnM2401 Actually I had another issue. I was testing out sending the redirect_uri parameter but that was causing an an error in vendor/league/oauth2-server/src/Grant/AbstractGrant.php in the validateClient method. 

When I removed the redirect_uri parameter from my request and edited validateClient inApi/V8/OAuth2/Repository/ClientRepository.php I was able to get an access_token from the refresh_token grant type.
